test/util/testfactory/utils.go

```diff
@@ -102,3 +109,89 @@ func GenerateAccounts(count int) []string {
 	}
 	return accs
 }
+
+func NewBaseAccount(kr keyring.Keyring, name string) (*authtypes.BaseAccount, sdk.Coins) {
+	if name == "" {
+		name = tmrand.Str(6)
+	}
+	rec, _, err := kr.NewMnemonic(name, keyring.English, "", "", hd.Secp256k1)
+	if err != nil {
+		panic(err)
+	}
+	addr, err := rec.GetAddress()
+	if err != nil {
+		panic(err)
+	}
+	origCoins := sdk.Coins{sdk.NewInt64Coin(bondDenom, BaseAccountDefaultBalance)}
+	bacc := authtypes.NewBaseAccountWithAddress(addr)
+	return bacc, origCoins
+}
+
+func GetValidators(grpcConn *grpc.ClientConn) (stakingtypes.Validators, error) {
+	scli := stakingtypes.NewQueryClient(grpcConn)
+	vres, err := scli.Validators(context.Background(), &stakingtypes.QueryValidatorsRequest{})
+
+	if vres == nil {
+		return stakingtypes.Validators{}, err
+	}
+	return vres.Validators, err
```

Comment:
yes it is, the reason is because `vres` can be nil and if it is nil, then `vres.Validators` would panic due to nil reference err.
we can probably go for something like this:
```go
if err != nil || vres == nil{
		return stakingtypes.Validators{}, err
}
```

Reply:
> the reason is because vres can be nil

Right but the critical part is if `vres` can be nil at the same time as `err`. If `vres` is nil I would expect `err` to be non-nil. In other words, this part should be sufficient:

```
if err != nil
		return stakingtypes.Validators{}, err
}
// do something with vres because it should be non-nil
```

It would be really confusing if the `Validators()` invocation returned `err = nil` AND `vres = nil` because that gives us no signal on if the method succeeded or failed.

There exists an example in this codebase of invoking `Validators()` that performs error handling how I would expect [here](https://github.com/celestiaorg/celestia-app/blob/3810531e027e05f76e137820572c766b1bcda1c3/test/txsim/stake.go#L104-L108)